test/integration/segmentation_camera.cc

```diff
@@ -0,0 +1,262 @@
+  camera->SetLocalRotation(0.0, 0.0, 0.0);
+
+  int backgroundLabel = 23;
+  camera->SetBackgroundLabel(backgroundLabel);
+  EXPECT_EQ(camera->BackgroundLabel(), backgroundLabel);
+
+  camera->SetSegmentationType(SegmentationType::ST_SEMANTIC);
+
+  EXPECT_EQ(camera->Type(), SegmentationType::ST_SEMANTIC);
+
+  camera->EnableColoredMap(false);
+  EXPECT_FALSE(camera->IsColoredMap());
+
+  int width = 320;
+  int height = 240;
+  double aspectRatio = width/height;
+
+  camera->SetAspectRatio(aspectRatio);
+  camera->SetImageWidth(width);
+  camera->SetImageHeight(height);
+  camera->SetHFOV(IGN_PI / 2);
+
+  // add camera to the scene
+  scene->RootVisual()->AddChild(camera);
+
+  // Set a callback on the  camera sensor to get a Segmentation camera frame
+  ignition::common::ConnectionPtr connection =
+      camera->ConnectNewSegmentationFrame(
+          std::bind(OnNewSegmentationFrame,
+          std::placeholders::_1, std::placeholders::_2, std::placeholders::_3,
+          std::placeholders::_4, std::placeholders::_5));
+  EXPECT_NE(nullptr, connection);
+
+  // Update once to create image
+  camera->Update();
+
+  // Semantic Test
+
+  // get the center of each box, the percentages locates the center
+  math::Vector2d leftProj(width * 0.25, height * 0.5);
+  math::Vector2d rightProj(width * 0.75, height * 0.5);
+  math::Vector2d middleProj(width * 0.5, height * 0.5);
+
+  // get their index in the buffer
+  uint32_t leftIndex = (leftProj.Y() * width + leftProj.X()) * 3;
+  uint32_t rightIndex = (rightProj.Y() * width + rightProj.X()) * 3;
+  uint32_t middleIndex = (middleProj.Y() * width + middleProj.X()) * 3;
+
+  // test
+  g_mutex.lock();
+  g_counter = 0;
+
+  // cast the unsigned char to unsigned int to read it
+  uint8_t leftLabel =   g_buffer[leftIndex];
+  uint8_t rightLabel =  g_buffer[rightIndex];
+  uint8_t middleLabel = g_buffer[middleIndex];
+
+  // check the label
+  EXPECT_EQ(leftLabel  , 1);
+  EXPECT_EQ(middleLabel, 2);
+  EXPECT_EQ(rightLabel , 1);
+
+  // check if the first pixel(background) = the background label
+  int background = g_buffer[0];
+  EXPECT_EQ(background, backgroundLabel);
+
+  g_mutex.unlock();
+
+  // Instance/Panoptic  Test
+  camera->SetSegmentationType(SegmentationType::ST_PANOPTIC);
+
+  // Update once to create image
+  camera->Update();
+
+  // the label in the last channel
+  leftLabel =   g_buffer[leftIndex   + 2];
+  rightLabel =  g_buffer[rightIndex  + 2];
+  middleLabel = g_buffer[middleIndex + 2];
+
+  // the instances count in the first channel
+  uint8_t leftCount =   g_buffer[leftIndex];
+  uint8_t rightCount =  g_buffer[rightIndex];
+  uint8_t middleCount = g_buffer[middleIndex];
+
+  // test
+  g_mutex.lock();
+  g_counter = 0;
+
+  // check the label
+  EXPECT_EQ(leftLabel  , 1);
+  EXPECT_EQ(middleLabel, 2);
+  EXPECT_EQ(rightLabel , 1);
+
+  // instance count
+  EXPECT_EQ(middleCount, 1);
+  EXPECT_EQ(rightCount, 1);
+  EXPECT_EQ(leftCount, 2);
```

Comment:
Can you add a comment that explains why `rightCount` is 1 and `leftCount` is 2 instead of it being the other way around? Is it due to the names of the left and right boxes, and how they're processed in the material switcher?